### PR TITLE
add tenzies

### PR DESCRIPTION
**Pull request by nngrz**
(Note: It's ready to review. Will add more personalized features in the future.)
Screenshot 2024-09-11 at 16 13 18
Screenshot 2024-09-11 at 16 13 50

### CODE REVIEW

File: tenzies/src/components/Die.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from "react"
+
+export default function Die(props) {
```
Comment (balinskia) [inline tenzies/src/components/Die.js:3]: please destruct props

File: tenzies/src/App.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    function allNewDice() {
+        const newDice = []
+        for (let i = 0; i < 10; i++) {
+            newDice.push(generateNewDie())
+        }
+        return newDice
+    }
+
+    function rollDice() {
+        if(!tenzies) {
```
Comment (balinskia) [inline tenzies/src/App.js:38]: there should be a space before "("

File: tenzies/src/App.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import React from "react"
+import Die from "./components/Die"
+import {nanoid} from "nanoid"
+import Confetti from 'react-confetti'
+
+export default function App() {
+    const [dice, setDice] = React.useState(allNewDice())
+    const [tenzies, setTenzies] = React.useState(false)
```
Comment (balinskia) [inline tenzies/src/App.js:8]: what does tenzies variable represent? Please add a comment

File: tenzies/src/App.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        const allSameValue = dice.every(die => die.value === dice[0].value)
+
+        if (allHeld && allSameValue) {
+            setTenzies(true)
+        }
+
+    }, [dice])
+
+    // this is a helper function to avoid having repetitive code
+    function generateNewDie() {
```
Comment (balinskia) [inline tenzies/src/App.js:21]: Who is dying?
I think it should be` generateNewDice`? (ending with `dice`, not `die`).

File: tenzies/src/App.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        const newDice = []
+        for (let i = 0; i < 10; i++) {
+            newDice.push(generateNewDie())
+        }
+        return newDice
+    }
+
+    function rollDice() {
+        if(!tenzies) {
+            setDice(oldDice => oldDice.map(die => {
```
Comment (balinskia) [inline tenzies/src/App.js:39]: die => dice

File: tenzies/src/App.js (hunk 4 above)
Comment (nngrz) [reply tenzies/src/App.js:21]: should be "die", die is singular

**Pull request merged**